[terra-Profile-image] added Accessibility Guide (#3684)
cerner/terra-core: packages/terra-core-docs/src/terra-dev-site/doc/profile-image/AccessibilityGuide.4.doc.mdx

</Notice>

## Accessibility Considerations

### Content Considerations: What do content creators need to do?

> - Content creators must ensure that images with meaningful content have a textual equivalent.
>   - Provide engineers with text for the “alt prop”.
>   - The alt prop must be a textual equivalent that conveys the meaning of the image.
> - For supplemental or decorative images that provide no contextual value, content creators must convey to the engineers that no alt prop will be supplied.
> - Content creators must work with engineers to convey any visual relationships created with Profile Image and other content on the page.

### Code Considerations: What do engineers need to do?

- Engineers must not override the Profile Image’s role of image.
- Engineers must apply an alt prop for meaningful images.
<div aria-label="Example Demo">
  <ProfileImageExamples />
</div>
```jsx
const cx = classNames.bind(styles);

const ProfileImageExamples = () => (
  <div>
    <Card>
      <Card.Body>
        <ProfileImage alt="" width="75" height="75" aria-describedby="caption-example-1" />

>         <span id="caption-example-1" className={cx('add-left-margin')}>Decorative Profile Image Source</span>

Reviewer: ```suggestion
        <span id="caption-example-1" className={cx('add-left-margin')}>Decorative Profile Image</span>
```

      </Card.Body>
    </Card>
    <Card>
      <Card.Body>
        <ProfileImage alt="profile image for Martin Moon, photograph from space of rising star over a planet's silhouette" src={exampleProfileImage} aria-describedby="caption-example-2" />

>         <span id="caption-example-2" className={cx('add-left-margin')}>Meaningful Profile Image Source</span>

Reviewer: ```suggestion
        <span id="caption-example-2" className={cx('add-left-margin')}>Meaningful Profile Image</span>
```

  - If so, the engineer must associate the Profile Image to the related content.

## Usability Expectations

The user expects to understand the purpose of Profile Image regardless of how they experience the content.

### Interaction Detail

Profile image is not interactable.

### Keyboard Expectations

Because Profile Image is not actionable, it does not receive keyboard focus.

## Support Compliance

Terra is committed to ensuring its components provide maximum accessibility. Terra provides the ability to make accessibile products. However, using Terra components will not automatically make a product accessible.

The final responsibility lies with the consumers of Terra components — ensuring proper usage, programmatic context where needed, the words used to label elements, engineers following correct implementation patterns, and authors crafting content that follows best practice guidance — to make a product fully accessible.

### Primary accessibility criteria for Profile Image

- [**1.1.1 Non-text Content**](https://www.w3.org/WAI/WCAG21/quickref/#non-text-content) - Partially Supports

>   - Terra Profile Image contains a meaningful alt tag. The ALT prop is listed as optional. However, it is unclear what a null value will result in code. All images or anything with role=image must have an ALT attribute.

Reviewer: ```suggestion
  - Terra Profile Image contains a meaningful alt tag. The ALT prop is listed as optional. whenever null or no value provided for alt prop Terra Profile Image will be presented as decorative for accessibility user.
  
**Note:** _All images or anything with role=image must have an ALT attribute_.
```

<div aria-label="Example Demo">
  <ProfileImageExamples />
</div>
> ```jsx
> const cx = classNames.bind(styles);
> 
> const ProfileImageExamples = () => (
>   <div>
>     <Card>
>       <Card.Body>
>         <ProfileImage alt="" width="75" height="75" aria-describedby="caption-example-1" />
>         <span id="caption-example-1" className={cx('add-left-margin')}>Decorative Profile Image Source</span>
>       </Card.Body>
>     </Card>
>     <Card>
>       <Card.Body>
>         <ProfileImage alt="profile image for Martin Moon, photograph from space of rising star over a planet's silhouette" src={exampleProfileImage} aria-describedby="caption-example-2" />
>         <span id="caption-example-2" className={cx('add-left-margin')}>Meaningful Profile Image Source</span>
>       </Card.Body>
>     </Card>
>   </div>
);
> ```
> - Work with content creators to understand how content should resize and reflow at different breakpoints and form factors to ensure content is not artificially constrained or cut off at smaller viewports.
> - Engineers must work with the content creators to understand if Profile Image is visual related to other content.

> >   - If so, the engineer must associate the Profile Image to the related content.

Reviewer 2: Remove the `>` from these lines.

## Usability Expectations

### Interaction Detail

Profile image is not interactable.

### Usability Expectations

> The user expects to understand the purpose of Profile Image regardless of how they experience the content.

Reviewer 2: Instead of having the usability expectations heading twice this section should be moved under the first heading iteself.

### Keyboard Expectations

Because Profile Image is not actionable, it does not receive keyboard focus.

## Support Compliance

> ### How does Profile Image support compliance?

Reviewer 2: This heading can be removed.

> Terra is committed to ensuring its components provide maximum accessibility. Terra provides the ability to make accessibility products. However, using Terra components will not automatically make a product accessible.

Reviewer 2: ```suggestion
Terra is committed to ensuring its components provide maximum accessibility. Terra provides the ability to make accessible products. However, using Terra components will not automatically make a product accessible.
```

The final responsibility lies with the consumers of Terra components — ensuring proper usage, programmatic context where needed, the words used to label elements, engineers following correct implementation patterns, and authors crafting content that follows best practice guidance — to make a product fully accessible.

### Primary accessibility criteria for Profile Image

- [**1.1.1 Non-text Content**](https://www.w3.org/WAI/WCAG21/quickref/#non-text-content) - Partially Supports

>   - Terra Profile Image contains a meaningful alt tag. The ALT prop is listed as optional. However, it is unclear what a null value will result in code. All images or anything with role=image must have an ALT attribute.  (UPDATE THIS SECTION WHEN TERRA FINISHED UPDATING)

Reviewer 2: What needs to be updated here? Please check with rahul/scott,.
```suggestion
  - Terra Profile Image contains a meaningful alt tag. The ALT prop is listed as optional. However, it is unclear what a null value will result in code. All images or anything with role=image must have an ALT attribute.  
```
